### ISSUE TITLE
Improvement on reimplemented-starmap rule (FURB140)

### ISSUE THREAD

**Issue by alexprengere** [astral-sh/ruff#7636]
I tested Ruff 0.0.291 which implements FURB140. It works great! I have a suggestion to cover more use cases in the same spirit:

```python
def average(*args):
    return sum(args) / len(args)

# Correctly reported
print(sum(average(a, b) for a, b in [(85, 60), (100, 80)]))

# Currently not reported
print(sum(average(*t) for t in [(85, 60), (100, 80)]))
```
